test/logger.test.js

```diff
@@ -124,3 +124,32 @@ test('expose the logger', t => {
   t.ok(fastify.logger)
   t.is(typeof fastify.logger, 'object')
 })
+
+test('The logger should accept a custom genReqId function', t => {
+  t.plan(3)
+
+  const fastify = Fastify({
+    logger: {
+      genReqId: function () {
+        return 'a'
+      }
+    }
+  })
+
+  fastify.get('/', (req, reply) => {
+    t.ok(req.req.id)
+    reply.send({ id: req.req.id })
```

Comment:
IMHO we should expose this in our wrapped request.

Reply:
Why?

Reply:
It seems an interesting info. Maybe not.

Reply:
We can discuss this once we add also the `header` object to request, it is more *in scope* discussion :)

Reply:
ok

Reply:
+1